course3/homework-unique_ptr/unique_ptr.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <cstddef>
+
+
+namespace karun
+{
+	template <class T>
+	class unique_ptr
+	{	
+		
+	public:
+		unique_ptr();
+		unique_ptr(std::nullptr_t);
```

Comment:
I forgot to ask this before, but even if I remove the `unique_ptr(std::nullptr_t)` constructor, calling `unique_ptr<int> p(nullptr);` creates a new unique_ptr p, with p.mPtr equal to nullptr. I was surprised by this, and expected the code to not compile since the only compatible constructor (`unique_ptr(T*)`) is marked as explicit - is nullptr a special case, able to implicitly cast to T* despite the explicit specifier?